### PR TITLE
Adnuntius: Add Price Type

### PR DESCRIPTION
**Pull request by mikael-lundin**
This PR will allow Prebid server to request a price type. The ad will be returned with either net, or gross bid depending on what you specify.

### CODE REVIEW

File: adapters/adnuntius/adnuntius.go [hunk 1]
```diff
@@ -344,23 +352,37 @@ func generateAdResponse(ad Ad, impId string, html string, request *openrtb2.BidR
...
+	}
+
 	adDomain := []string{}
 	for _, url := range ad.DestinationUrls {
 		domainArray := strings.Split(url, "/")
 		domain := strings.Replace(domainArray[2], "www.", "", -1)
 		adDomain = append(adDomain, domain)
 	}
 
+	// strings.ToLower(str)
```
Comment (onkarvhanumante) [inline adapters/adnuntius/adnuntius.go:374]: nitpick: should remove comment

File: openrtb_ext/imp_adnuntius.go [hunk 2]
```diff
@@ -5,4 +5,5 @@ type ImpExtAdnunitus struct {
 	Network   string `json:"network"`
 	NoCookies bool   `json:"noCookies"`
 	MaxDeals  int    `json:"maxDeals"`
+	PriceType string `json:"priceType,omitempty"`
```
Comment (onkarvhanumante) [inline openrtb_ext/imp_adnuntius.go:8]: should also update 
https://github.com/prebid/prebid-server/blob/71f76eb39e23d778e7ff3356711004775851e56e/static/bidder-params/adnuntius.json

to include `priceType` as optional param

Comment (onkarvhanumante) [inline openrtb_ext/imp_adnuntius.go:8]: update [params tests](https://github.com/prebid/prebid-server/blob/master/adapters/adnuntius/params_test.go) to add valid and invalid tests for `PriceType`

**Comment by mikael-lundin** [comment]
Thanks for the feedback, I've changed it and pushed to this PR.

**Comment by onkarvhanumante** [comment]
@mikael-lundin  should create docs PR to include details for PriceType and maxDeals

File: adapters/adnuntius/adnuntius.go [hunk 3]
```diff
@@ -39,6 +39,14 @@ type Ad struct {
 		Amount   float64
 		Currency string
 	}
+	NetBid struct {
+		Amount   float64
+		Currency string
+	}
+	GrossBid struct {
+		Amount   float64
+		Currency string
```
Comment (onkarvhanumante) [inline adapters/adnuntius/adnuntius.go:48]: @mikael-lundin  could you help to understand how values are set to `NetBid.Currency` and `GrossBid.Currency`. If these vars are not in use then can should remove them

Comment (mikael-lundin) [reply adapters/adnuntius/adnuntius.go:48]: They are always returned from the ad-server so they will always be included in the response.

Comment (mikael-lundin) [reply adapters/adnuntius/adnuntius.go:48]: But yeah you are right they can be removed. :)

File: adapters/adnuntius/adnuntius.go [hunk 4]
```diff
@@ -344,6 +350,18 @@ func generateAdResponse(ad Ad, impId string, html string, request *openrtb2.BidR
 		}}
 	}
 
+	price := ad.Bid.Amount
+	priceType, _, _, _ := jsonparser.Get(imp.Ext, "bidder", "priceType")
```
Comment (SyntaxNode) [inline adapters/adnuntius/adnuntius.go:354]: Please do not ignore errors.

File: adapters/adnuntius/adnuntius.go [hunk 5]
```diff
@@ -344,6 +350,18 @@ func generateAdResponse(ad Ad, impId string, html string, request *openrtb2.BidR
...
+
+	if string(priceType) != "" {
+		if strings.ToLower(string(priceType)) == "net" {
```
Comment (SyntaxNode) [inline adapters/adnuntius/adnuntius.go:357]: Please prefer to use `strings.EqualFold` for a case insensitive comparison.

**Comment by mikael-lundin** [comment]
I'm waiting for this PR to go through before adding the prebid server things:
https://github.com/prebid/prebid.github.io/pull/4849/files

Is it ok to have one name for prebid client and another for prebid server? it seems when I looked into the documentation that server prefered the name "priceType"

File: adapters/adnuntius/adnuntius.go (hunk 5 above)
Comment (mikael-lundin) [reply adapters/adnuntius/adnuntius.go]: Thanks, this is changed.

**Comment by onkarvhanumante** [comment]
> I'm waiting for this PR to go through before adding the prebid server things: https://github.com/prebid/prebid.github.io/pull/4849/files
> 
> Is it ok to have one name for prebid client and another for prebid server? it seems when I looked into the documentation that server prefered the name "priceType"

assuming prebid.js is being referred as prebid client. It would be better if both prebid js and prebid server use same name for new bidder info param.

**Comment by mikael-lundin** [comment]
Good to know, I will make change the name in prebid server as well.

**Comment by onkarvhanumante** [comment]
> Good to know, I will make change the name in prebid server as well.

@mikael-lundin any updates to move this PR forward

**Comment by mikael-lundin** [comment]
Yes, will be done today! sorry for the wait.

File: static/bidder-params/adnuntius.json [hunk 6]
```diff
@@ -9,13 +9,21 @@
...
             "type": "string",
             "description": "Network if required"
         },
         "noCookies": {
             "type": "boolean",
             "description": "Disable cookies being set by the ad server."
+        },
+        "priceType": {
+            "type": "string",
+            "description": "Allows you to specify Net or Gross bids."
```
Comment (onkarvhanumante) [inline static/bidder-params/adnuntius.json:26]: should update `priceType` to `bidType`

Comment (mikael-lundin) [reply static/bidder-params/adnuntius.json]: My bad .:/

**Review by onkarvhanumante — APPROVED** [review]

**Review by gargcreation1992 — APPROVED** [review]

**Pull request merged**